packages/build/src/steps/run_core_step.js

```diff
@@ -0,0 +1,209 @@
+/* eslint-disable max-lines */
+import { getConfigOpts, loadConfig } from '../core/config.js'
+import { getConstants } from '../core/constants.js'
+import { normalizeFlags } from '../core/normalize_flags.js'
+import { getSeverity } from '../core/severity.js'
+import { handleBuildError } from '../error/handle.js'
+import { getErrorInfo } from '../error/info.js'
+import { startErrorMonitor } from '../error/monitor/start.js'
+import { getBufferLogs } from '../log/logger.js'
+import { logBuildStart } from '../log/messages/core.js'
+import { bundleEdgeFunctions } from '../plugins_core/edge_functions/index.js'
+import { reportStatuses } from '../status/report.js'
+
+import { runSteps } from './run_steps.js'
+
+/**
+ * Executes specified build steps for functions
+ *
+ * @param  {object} [flags] - build configuration CLI flags
+ * @param  {string} [flags.config] - Path to the configuration file
+ * @param  {string} [flags.cwd] - Current directory. Used to retrieve the configuration file
+ * @param  {string} [flags.repositoryRoot] - Git repository root directory. Used to retrieve the configuration file.
+ * @param  {string} [flags.apiHost] - Netlify API endpoint
+ * @param  {string} [flags.token] - Netlify API token for authentication
+ * @param  {string} [flags.siteId] - Netlify Site ID
+ * @param  {string} [flags.deployId] - Netlify Deploy ID
+ * @param  {string} [flags.context] - Build context
+ * @param  {string} [flags.branch] - Repository branch
+ * @param  {boolean} [flags.dry=false] - Run in dry mode, i.e. printing steps without executing them
+ * @param  {string} [flags.nodePath] - Path to the Node.js binary to use in the build command and plugins
+ * @param  {boolean} [flags.buffer=false] - Buffer output instead of printing it
+ *
+ * @returns {object} buildResult
+ * @returns {boolean} buildResult.success - Whether build succeeded or failed
+ * @returns {number} buildResult.severityCode - Build success/failure status among:
+ * 0 (success), 1 (build cancelled), 2 (user error), 3 (plugin error), 4 (system error). Can be used as exit code.
+ * @returns {string[]} buildResult.logs - When using the `buffer` option, all log messages
+ */
+export default async function runCoreStep(buildSteps, flags = {}) {
+  const { errorMonitor, mode, logs, debug, ...flagsA } = startBuild(flags)
+
+  const errorParams = { errorMonitor, mode, logs, debug }
+
+  try {
+    const { netlifyConfig: netlifyConfigA, configMutations } = await executeBuildStep({
+      ...flagsA,
+      errorMonitor,
+      mode,
+      logs,
+      debug,
+      errorParams,
+      buildSteps,
+    })
+
+    const { success, severityCode } = getSeverity('success')
+
+    return { success, severityCode, netlifyConfig: netlifyConfigA, configMutations, logs }
+  } catch (error) {
+    const { severity } = await handleBuildError(error, errorParams)
+
```

Comment:
nit: Removing empty line between declarations inside method, to reduce verbosity and keep consistency with the rest of the codebase.

```suggestion
```